app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -0,0 +1,51 @@
+module ForemanDeployments
+  module Tasks
+    class SearchTaskDefinition < BaseDefinition
+      class Action < BaseAction
+        def run
+          results = SearchTaskDefinition.create_object(input)
+          results = SerializableArray.new(results)
+          SearchTaskDefinition.create_output(results, output)
+        end
+      end
+
+      class SerializableArray < Array
+        def to_hash
+          {}
+        end
+      end
+
+      def validate
+        results = SearchTaskDefinition.create_object(parameters)
+        messages = nil
+        unless results.any?
+          messages = ["#{parameters[:klass]}.search_for(\"#{parameters[:search_term]}\") didnt return valid objects"]
+        end
+
+        ValidationResult.new(messages)
+      end
+
+      def preliminary_output(parameters)
+        SearchTaskDefinition.create_output(SearchTaskDefinition.create_object(parameters))
+      end
+
+      def dynflow_action
+        ForemanDeployments::Tasks::SearchTaskDefinition::Action
+      end
+
+      def self.create_object(parameters)
+        object_type = parameters['klass']
+        object_type = object_type.constantize if object_type.is_a? String
+        results = object_type.search_for(parameters[:search_term])
```

Comment:
The task always selects all available records. The search term is in fact stored in `parameters['params']['search_term']`. The code above always calls `.search_for(nil)` which returns all available.